packages/mui-material-next/src/Slider/Slider.tsx

```diff
@@ -0,0 +1,1090 @@
+};
+
+export { SliderTrack };
+
+const SliderThumb = styled('span', {
+  name: 'MuiSlider',
+  slot: 'Thumb',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+    return [
+      styles.thumb,
+      styles[`thumbColor${capitalize(ownerState.color || 'primary')}`],
+      ownerState.size !== 'medium' && styles[`thumbSize${capitalize(ownerState.size)}`],
+    ];
+  },
+})<{ ownerState: SliderOwnerState }>(({ theme, ownerState }) => {
+  const { vars: tokens } = theme;
+
+  function getBoxShadow(state: keyof MD3State) {
+    return `0px 0px 0px 10px rgba(${tokens.sys.color.primaryChannel} / ${tokens.sys.state[state].stateLayerOpacity})`;
+  }
+
+  return {
+    position: 'absolute',
+    width: 20,
+    height: 20,
+    boxSizing: 'border-box',
+    borderRadius: tokens.sys.shape.corner.full,
+    outline: 0,
+    backgroundColor: 'currentColor',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    transition: theme.transitions.create(['box-shadow', 'left', 'bottom'], {
+      duration: theme.transitions.duration.shortest,
+    }),
+    ...(ownerState.size === 'small' && {
+      width: 12,
+      height: 12,
+    }),
+    ...(ownerState.orientation === 'horizontal' && {
+      top: '50%',
+      transform: 'translate(-50%, -50%)',
+    }),
+    ...(ownerState.orientation === 'vertical' && {
+      left: '50%',
+      transform: 'translate(-50%, 50%)',
+    }),
+    '&:before': {
+      position: 'absolute',
+      content: '""',
+      borderRadius: 'inherit',
+      width: '100%',
+      height: '100%',
+      boxShadow: tokens.sys.elevation[1],
+      ...(ownerState.size === 'small' && {
+        boxShadow: 'none',
+      }),
+    },
+    '&::after': {
+      position: 'absolute',
+      content: '""',
+      borderRadius: '50%',
+      // 40px is the hit target
+      width: 40,
+      height: 40,
+      top: '50%',
+      left: '50%',
+      transform: 'translate(-50%, -50%)',
+    },
+    [`&:hover`]: {
+      boxShadow: getBoxShadow('hover'),
+      '@media (hover: none)': {
+        boxShadow: 'none',
+      },
+    },
+    [`&.${sliderClasses.focusVisible}`]: {
+      boxShadow: getBoxShadow('focus'),
+    },
+    [`&.${sliderClasses.active}`]: {
+      boxShadow: getBoxShadow('pressed'),
+    },
+    [`&.${sliderClasses.disabled}`]: {
+      boxShadow: 'none',
+      '&:before': {
+        boxShadow: tokens.sys.elevation[0],
+      },
+    },
+  };
+});
+
+SliderThumb.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit the d.ts file and run "yarn proptypes"     |
+  // ----------------------------------------------------------------------
+  /**
+   * @ignore
+   */
+  children: PropTypes.node,
+};
+
+export { SliderThumb };
+
+const StyledSliderValueLabel = styled(SliderValueLabel, {
```

Comment:
Interesting, did you copy the implementation of Material UI v4 https://github.com/mui/material-ui/blob/v4.x/packages/material-ui/src/Slider/ValueLabel.js#L5 https://v4.mui.com/components/slider/#discrete-sliders or from somewhere else? To see if there aren't things to copy from there.

It seems that with MD v3, they decided to revert the value label of MD v2 https://m2.material.io/components/sliders#usage back to MD v1 https://m1.material.io/components/sliders.html#sliders-discrete-slider 🙃.

Reply:
Didn't know MD v1 had the same value label 😅. I copied everything from Material UI v5, and then modified the styles on top of it. For the value label, I actually used [this Base UI example](https://mui.com/base/react-slider/#value-label) as a guide as it has the same type of value label.